### PR TITLE
Use hqDefine in new export code

### PR DESCRIPTION
**Pull request by dannyroberts**
@NoahCarnahan @benrudolph

### CODE REVIEW

**Comment by dannyroberts** [comment]
Curious what you all think; @benrudolph it looks like you were making a great effort to be responsible with your global variables here before we'd introduced `hqDefine`. It was a pretty clean switch tu using `hqDefine`.

Also, if there's ongoing QA on new exports code, would it be worth getting this change QA'd in the same pass?

**Comment by dannyroberts** [comment]
cc: @gcapalbo

File: corehq/apps/export/templates/export/customize_export_new.html [hunk 1]
```diff
@@ -142,21 +141,21 @@
             <div class="col-sm-offset-4 col-md-offset-3 col-lg-offset-2 col-sm-8 col-md-9 col-lg-10 controls">
                 <button type="submit" class="btn btn-lg btn-primary" data-bind="
                     click: save,
-                    disable: (saveState() === Exports.Constants.SAVE_STATES.SAVING ||
-                              saveState() === Exports.Constants.SAVE_STATES.SUCCESS)
+                    disable: (saveState() === hqImport('export/js/const.js').SAVE_STATES.SAVING ||
+                              saveState() === hqImport('export/js/const.js').SAVE_STATES.SUCCESS)
```
Comment (gcapalbo) [inline corehq/apps/export/templates/export/customize_export_new.html]: is the return value of hqImport memoized within a single page load?

**Comment by gcapalbo** [comment]
build failure looks related:

    Running "mocha:export#ko" (mocha) task
    Testing: http://172.17.0.12:8000/mocha/export/ko
    Warning: Error: The module 'export/js/utils.js' has not yet been defined.
        Did you include <script src="export/js/utils.js"></script> on your html page?
        at hqImport (http://172.17.0.12:8000/static/hqwebapp/js/hqModules.js:51)
        at undefined (http://172.17.0.12:8000/static/export/js/models.js:5)
        at hqDefine (http://172.17.0.12:8000/static/hqwebapp/js/hqModules.js:45)
        at undefined (http://172.17.0.12:8000/static/export/js/models.js:292)
    Use --force to continue.
    Aborted due to warnings.

File: corehq/apps/export/templates/export/customize_export_new.html (hunk 1 above)
Comment (dannyroberts) [inline corehq/apps/export/templates/export/customize_export_new.html]: `hqImport` as a function just looks up a memoized value; behind the scenes `hqDefine` adds a value to a global object and `hqImport` looks it up from the global object (while adding some error checking and other lightweight niceties). I wouldn't worry about calling `hqImport` being slow.

Comment (gcapalbo) [inline corehq/apps/export/templates/export/customize_export_new.html]: cool

**Comment by NoahCarnahan** [comment]
Looks great Danny. The QA is ongoing (behind a feature flag), so I would suggest we merge this once the tests are passing. The remaining QA plus the re-testing they do for issues they've already found should be sufficient to test this I think.

**Comment by dimagimon** [comment]
There are 22 errors:

* corehq/apps/export/static/export/js/models.js, line 69 - 'analytics' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 172 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 76 - 'analytics' is not defined. (no-undef)
* corehq/apps/export/static/export/js/models.js, line 114 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 80 - 'analytics' is not defined. (no-undef)
* corehq/apps/export/static/export/js/const.js, line 19 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 113 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 210 - 'key' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/const.js, line 25 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 1 - 'Exports' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/models.js, line 167 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 283 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 250 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/const.js, line 30 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 22 - 'key' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/models.js, line 274 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 173 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/const.js, line 34 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/js/models.js, line 58 - 'response' is defined but never used (no-unused-vars)
* corehq/apps/export/static/export/js/models.js, line 81 - 'analytics' is not defined. (no-undef)
* corehq/apps/export/static/export/js/utils.js, line 18 - Missing trailing comma. (comma-dangle)
* corehq/apps/export/static/export/spec/ExportInstance.spec.js, line 59 - 'instanceSaved' is not defined. (no-undef)
'savedFormExport' is not defined. (no-undef)

**Comment by dannyroberts** [comment]
OK, just appeased eslint and tests are passing. I haven't done manual QA, but it sounds like we're ok lumping it in with the current QA.

**Comment by NoahCarnahan** [comment]
I tested this locally by loading an export configuration page (with the flag enabled) and looks like the are some errors (same as from mocha test).

**Comment by dannyroberts** [comment]
Thanks Noah. I just tested locally and found one more bug as well. Feel free to take another pass on your machine, but I think this is good now.

**Comment by NoahCarnahan** [comment]
👍 on ✅

**Comment by dannyroberts** [comment]
✅

**Comment by benrudolph** [comment]
nice thanks

**Pull request merged**